fix(urlencode): urlencode CIRCLE_BRANCH [semver:patch]

andrew-barnett opened the pull request:
### Motivation, issues

The CircleCI API (started?) requiring urlencoded branch names, otherwise the calls were returning empty results. This caused the queue to never succeed. This fix urlencodes the branch name. See #89

### Description

add `urlencode` function and run `${CIRCLE_BRANCH}` through it while building branch-specific API url

fixes #89

andrew-barnett committed:
fix(urlencode): urlencode CIRCLE_BRANCH

fixes #eddiewebb/circleci-queue#89

eddiewebb commented on src/commands/until_front_of_line.yml:
```diff
@@ -51,6 +51,19 @@ steps:
           tag_pattern=${tag_pattern:1:-1}
         fi
 
+        urlencode(){
```
@andrew-barnett - how much of this is specific on a shell (bash, zsh, ash)? Or is it portable?

eddiewebb commented:
bors try

eddiewebb commented:
BotComment: *Development* version of orb available for manual validation - `eddiewebb/queue@dev:417`

eddiewebb commented:
bors delegate+

eddiewebb commented:
@andrew-barnett - please see my comment/concern about portability.  Assuming it works across those 3 shells I have delegated review permission, so you can just add a comment here of `bors merge`

andrew-barnett commented:
@eddiewebb I'm not finding your comment about which shells to check. Can you post here?

eddiewebb commented:
I'm sorry Andrew it was stuck pending, submitted it.

andrew-barnett committed:
fix(urlencode): update urlencode for portability

changed how string is iterated over to support zsh.
tested on bash, zsh and the flavor of sh that's on cimg/base:stable

andrew-barnett commented:
> @andrew-barnett - please see my comment/concern about portability. Assuming it works across those 3 shells I have delegated review permission, so you can just add a comment here of `bors merge`

@eddiewebb - It didn't work on zsh, so updated the code to make it portable. Passing it back for review.

andrew-barnett commented:
bors try

eddiewebb commented:
BotComment: *Development* version of orb available for manual validation - `eddiewebb/queue@dev:421`

andrew-barnett commented:
bors r+

eddiewebb commented:
BotComment: *Development* version of orb available for manual validation - `eddiewebb/queue@dev:424`

eddiewebb committed:
changelog

eddiewebb committed:
addd semver check

eddiewebb commented:
bors merge

eddiewebb commented:
( ialways get the magic PR title wrong, added a sanity check and changelog)

Version 1.8.2 and soon 1.8.3 wil include this fix, thanks @andrew-barnett !

eddiewebb committed:
Merge branch 'main' into issue-89-urlencode

eddiewebb commented:
bors merge

eddiewebb commented:
BotComment: *Development* version of orb available for manual validation - `eddiewebb/queue@dev:431`

eddiewebb commented:
BotComment: *Production* version of orb available for use - `eddiewebb/queue@1.8.4`
